src/java/picard/analysis/CollectGcBiasMetrics.java

```diff
@@ -101,107 +83,75 @@
     @Option(doc = "For summary metrics, exclude GC windows that include less than this fraction of the genome.")
     public double MINIMUM_GENOME_FRACTION = 0.00001;
 
-    @Option(doc = "If true, assume that the input file is coordinate sorted, even if the header says otherwise.",
-            shortName = StandardOptionDefinitions.ASSUME_SORTED_SHORT_NAME)
-    public boolean ASSUME_SORTED = false;
-
     @Option(shortName = "BS", doc = "Whether the SAM or BAM file consists of bisulfite sequenced reads.  ")
     public boolean IS_BISULFITE_SEQUENCED = false;
 
-    private static final Log log = Log.getInstance(CollectGcBiasMetrics.class);
-
     // Used to keep track of the total clusters as this is kinda important for bias
     private int totalClusters = 0;
     private int totalAlignedReads = 0;
+    // Histograms to track the number of windows at each GC, and the number of read starts
+    // at windows of each GC
+    final int[] windowsByGc = new int[101];
+    final int[] readsByGc = new int[101];
+    final long[] basesByGc = new long[101];
+    final long[] errorsByGc = new long[101];
+    static int lastContig = -1;
```

Comment:
there's a constant in SamRecord called NO_ALIGNMENT_REFERENCE_INDEX. Use this instead of defining your own.